WordPress/Classes/Networking/AccountServiceRemoteREST.m

```diff
@@ -151,6 +151,27 @@ - (void)isEmailAvailable:(NSString *)email success:(void (^)(BOOL available))suc
           }];
 }
 
+- (void)checkUsernameAvailability:(NSString *)username
+                            taken:(void (^)())taken
+                        available:(void (^)())available;
+{
+    NSString *path = @"https://public-api.wordpress.com/is-available/username";
+    [self.wordPressComRestApi GET:path
+                       parameters:@{ @"q": username, @"format": @"json"}
+                          success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
+                              if (taken) {
+                                  taken();
+                              }
+                          } failure:^(NSError *error, NSHTTPURLResponse *httpResponse) {
+                              // If the username is not available (has already been used)
+                              // the endpoint will reply with a 200 status code and an JSON
+                              // object describing an error.
```

Comment:
This would swallow any error, not just an error resulting from a non JSON response.